### DISCUSSION TITLE
QuitRequest

### DISCUSSION THREAD

**Discussion by fenikssc** [TempoDev/Arcade-Toulouse#18] [Ideas]
Ajouter une methode isRunning() pour arreter la game loop dans le core ?

**Comment by Pebloop** [comment]
Je comprends pas trop ce que tu veux que cette méthode fasse ?

**Reply by fenikssc** [reply]
Elle permettrait de savoir quand le jeu s'est arrêté et donc de repasser au menu

**Reply by nasso** [reply]
Pourquoi il y aurai besoin de faire apparaître cette méthode sur notre interface? C'est propre au fonctionnement interne de ton Arcade, nos interfaces ne relève que de la communication entre les librairies.

**Reply by Pebloop** [reply]
Je pense que ce qu'il veut globalement c'est une méthode exit() pour quitter le jeu.

**Reply by fenikssc** [reply]
Je ne vois pas comment le jeu d'un autre groupe va s'arrêter dans mon code et avec ma lib graphique,il faut bien une condition à la game loop non ?

**Reply by nasso** [reply]
Je pense que c'est plutôt à la lib graphique de donner l'information au core (par exemple si l'utilisateur ferme la fenêtre en cliquant sur la croix).

Dans notre POC, on a quelque chose comme ça:

```cpp
class IGraphic {
public:
    virtual ~IGraphic() = default;
    
    virtual bool quitRequested() const = 0;
};
```

`IGraphic::quitRequested` retourne `true` si l'utilisateur à cliqué sur la croix (ou a demandé à quitter d'une manière ou d'une autre).

**Reply by Pebloop** [reply]
Heu non pour le coup, parce que c'est pas ça le problème, si tu as un bouton quitter le jeu dans ton jeu pour retourner au menu (je parle du menu du core, hein), c'est dans ton game qu'il va être codé, d'où le besoin d'une methode exit() ou quelque chose du genre pour demander au core de quitter le jeu.

Par contre je ne pense pas qu'une condition soient nécessaire ? Juste une méthode qui réclame la fermeture du jeu au core quand executé.

**Reply by fenikssc** [reply]
Et bien les deux sont bon je pense, le côté graphique pour fermer la fenêtre et le côté game pour arrêter le jeu

**Reply by nasso** [reply]
Je pense aussi qu'il faudrait les deux

**Comment by nasso** [comment]
Tu devrais renommer le titre de cette discussion pour éviter de la confondre avec d'autres discussions qui parlent de l'interface `IGame`

**Comment by Pebloop** [comment]
Voir :  https://github.com/TempoDev/Arcade-Toulouse/issues/23 et https://github.com/TempoDev/Arcade-Toulouse/issues/22
